app/models/concerns/hyrax/ability/admin_set_ability.rb

```diff
@@ -2,28 +2,52 @@
           can :manage_any, AdminSet
+          can :manage_any, Hyrax::AdministrativeSet
           can :create_any, AdminSet
+          can :create_any, Hyrax::AdministrativeSet
           can :view_admin_show_any, AdminSet
+          can :view_admin_show_any, Hyrax::AdministrativeSet
         else
-          can :manage_any, AdminSet if Hyrax::Collections::PermissionsService.can_manage_any_admin_set?(ability: self)
-          can [:create_any, :create], AdminSet if Hyrax::CollectionTypes::PermissionsService.can_create_admin_set_collection_type?(ability: self)
-          can :view_admin_show_any, AdminSet if Hyrax::Collections::PermissionsService.can_view_admin_show_for_any_admin_set?(ability: self)
-
-          can [:edit, :update, :destroy], AdminSet do |admin_set| # for test by solr_doc, see solr_document_ability.rb
+          if Hyrax::Collections::PermissionsService.can_manage_any_admin_set?(ability: self)
+            can :manage_any, AdminSet
+            can :manage_any, Hyrax::AdministrativeSet
+          end
+          if Hyrax::CollectionTypes::PermissionsService.can_create_admin_set_collection_type?(ability: self)
+            can :create, [AdminSet, Hyrax::AdministrativeSet]
+            can :create_any, AdminSet
+            can :create_any, Hyrax::AdministrativeSet
+          end
+          if Hyrax::Collections::PermissionsService.can_view_admin_show_for_any_admin_set?(ability: self)
+            can :view_admin_show_any, AdminSet
+            can :view_admin_show_any, Hyrax::AdministrativeSet
+          end
+          # [:edit, :update, :destroy] for AdminSet is controlled by Hydra::Ability #edit_permissions
+          can [:edit, :update, :destroy], Hyrax::AdministrativeSet do |admin_set| # for test by solr_doc, see solr_document_ability.rb
             test_edit(admin_set.id)
           end
 
           can :deposit, AdminSet do |admin_set| # for test by solr_doc, see collection_ability.rb
             Hyrax::Collections::PermissionsService.can_deposit_in_collection?(ability: self, collection_id: admin_set.id)
           end
+          can :deposit, Hyrax::AdministrativeSet do |admin_set| # for test by solr_doc, see collection_ability.rb
+            Hyrax::Collections::PermissionsService.can_deposit_in_collection?(ability: self, collection_id: admin_set.id)
+          end
 
           can :view_admin_show, AdminSet do |admin_set| # admin show page # for test by solr_doc, see collection_ability.rb
             Hyrax::Collections::PermissionsService.can_view_admin_show_for_collection?(ability: self, collection_id: admin_set.id)
           end
+          can :view_admin_show, Hyrax::AdministrativeSet do |admin_set| # admin show page # for test by solr_doc, see collection_ability.rb
+            Hyrax::Collections::PermissionsService.can_view_admin_show_for_collection?(ability: self, collection_id: admin_set.id)
+          end
+
+          # [:read] for AdminSet is controlled by Hydra::Ability #read_permissions
```

Comment:
The check for [:read] for AF AdminSet was removed because the check already happens in Hydra::Ability.